### DISCUSSION TITLE
How to use PreCondition with ProjectTo

### DISCUSSION THREAD

**Discussion by cuongdo926** [LuckyPennySoftware/AutoMapper#3991] [NA]
I have a PostDto class that has some count properties such as CommentCount, and AnswerCount...
How can I use ProjectTo with PreCondition or is there any way to do it with AutoMapper? 

```
// query
PostSet.AsNoTracking().ProjectTo<PostDto>({ postType : 'Post' })
PostSet.AsNoTracking().ProjectTo<PostDto>({ postType : 'Question' })

// map configuration
string postType = null;
configuration.CreateMap<Post, PostDto>()
    .ForMember(x => x.AnswerCount, opt =>
    {
        opt.PreCondition((Post _) => postType == 'Post);
        opt.MapFrom(x => x.Answers.Count());
    })
    .ForMember(x => x.CommentCount, opt =>
    {
        opt.PreCondition((Post _) => postType == 'Question');
        opt.MapFrom(x => x.Comments.Count());
    })
    ;
```

Below are 2 worked solutions but I wonder if there is any way to use with AutoMapper:
- Add different DTO classes with exact properties used for ProjectTo
- Project with same DTO class using ```Select``` but the code is messy with many manual map and many conditions to get proper ```Select```

**Comment by lbargaoanu** [comment]
https://docs.automapper.org/en/latest/Queryable-Extensions.html#supported-mapping-options
